projects/plugins/jetpack/modules/calypsoify/class-jetpack-calypsoify.php

```diff
@@ -75,33 +75,58 @@ public function setup() {
 
 		if ( $this->is_calypsoify_enabled ) {
 			add_action( 'admin_init', array( $this, 'setup_admin' ), 6 );
-			add_action( 'admin_menu', array( $this, 'remove_core_menus' ), 100 );
-			add_action( 'admin_menu', array( $this, 'add_custom_menus' ), 101 );
+
+			if ( $this->should_override_nav() ) {
+				add_action( 'admin_menu', array( $this, 'remove_core_menus' ), 100 );
+				add_action( 'admin_menu', array( $this, 'add_custom_menus' ), 101 );
+			}
```

Comment:
Could you explain why we don't place `setup_admin` behind the `should_override_nav` check here already? Is this because we want to enqueue the Gutenberg scripts at all times? Would it be an option to move those enqueues here instead, to make it easier to parse what Calypsoify features are available and when?

Reply:
> Could you explain why we don't place `setup_admin` behind the `should_override_nav` check here already? Is this because we want to enqueue the Gutenberg scripts at all times?

Partially. It's because we need to perform the `is_page_gutenberg` check at all times in order to determine if the Gutenberg scripts should be enqueued, since those customizations are independent from the navigation customizations.

> Would it be an option to move those enqueues here instead, to make it easier to parse what Calypsoify features are available and when?

I tried that, but the `is_page_gutenberg` check uses `use_block_editor_for_post_type` which is only available after the `admin_init` hook.